spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/PercentageBasedSampler.java

```diff
@@ -40,7 +38,7 @@ public boolean next(Void info) {
 		if (currentSpan == null) {
 			return false;
 		}
-		return new UuidTraceIdToThresholdComparable(configuration.getPercentage(), converter)
+		return new TraceIdToThresholdComparable(configuration.getPercentage())
```

Comment:
just inline this. ex. if `configuration.getPercentage()` is mutable config, make a final copy. This code is simple enough that it can be inlined, and there's no reason to require an allocation or another type to do this math.